src/plugins/lua/replies.lua

```diff
@@ -34,42 +34,107 @@ local settings = {
+  reply_sender_privacy_length = 16,
 }
 
 local N = "replies"
 
 local function make_key(goop, sz, prefix)
   local h = hash.create()
   h:update(goop)
-  local key
-  if sz then
-    key = h:base32():sub(1, sz)
-  else
-    key = h:base32()
-  end
+  local key = (prefix or '') .. h:base32():sub(1, sz)
+  return key
+end
 
-  if prefix then
-    key = prefix .. key
-  end
+local global_replies_set_script
+local local_replies_set_script
+
+local function configure_redis_scripts(_, _)
+  local redis_script_zadd_global = [[
+      -- getting last score of the recipients from the global replies set
+      local data = {}
+      data = redis.call('ZRANGE', KEYS[1], '-1', '-1', 'WITHSCORES')
+      local score = tonumber(data[#data])
+      if score == nil then
+        score = 0
+      end
 
-  return key
+      local global_size = redis.call('ZCARD', KEYS[1])
+      if global_size > {= mgs =} then
+        redis.call('ZREMRANGEBYRANK', KEYS[1], '{= mgs =}', global_size)
+      end
+
+      local recipients_addrs = ARGV
+      if recipients_addrs ~= nil then
+        local params = {}
+        --table.insert(params, KEYS[1])
+        for _, rcpt in ipairs(recipients_addrs) do
+          -- score is using here as a one-based numeration just like in lua
+          score = score + 1
+          table.insert(params, score)
+          table.insert(params, tostring(rcpt))
+        end
+        -- adding recipients to the global replies set
+        redis.call('ZADD', KEYS[1], unpack(params))
+      end
+      ]]
+  local set_script_zadd_global = lua_util.jinja_template(redis_script_zadd_global,
+          { mgs = settings['max_global_size'] })
+  rspamd_logger.debugm(N, rspamd_config, 'added add to global replies set script %s', set_script_zadd_global)
+  global_replies_set_script =  lua_redis.add_redis_script(set_script_zadd_global, redis_params)
+
+  local redis_script_zadd_local = [[
+      local local_replies_set_size = redis.call('ZCARD', KEYS[1])
+      if local_replies_set_size > {= mls =} then
+        redis.call('ZREMRANGEBYRANK', KEYS[1], '{= mls =}', local_replies_set_size)
+      end
+      local given_params = ARGV
+      if given_params ~= nil then
+          local task_time = given_params[1]
+          table.remove(given_params, 1)
+          -- passing_params is a table that will be passed to the redis
+          local passing_params = {}
+          for _, rcpt in ipairs(given_params) do
+            -- adding recipients for the local replies set
+            table.insert(passing_params, task_time)
+            table.insert(passing_params, rcpt)
+          end
+          redis.call('ZADD', KEYS[1], unpack(passing_params))
+
+          -- setting expire for local replies set
+          redis.call('EXPIRE', KEYS[1], tostring(math.floor('{= exp =}')))
+      end
+    ]]
+  local set_script_zadd_local = lua_util.jinja_template(redis_script_zadd_local,
+          { exp = settings['expire'], mls = settings['max_local_size'] })
```

Comment:
```suggestion
          { exp = settings.expire, mls = settings.max_local_size })
```

NIT: use `.` to access constant literal named elements in a table.

Reply:
Also, neither `mls` nor `exp` do not look like a good variable names to me. Why not use `expire` and `max_local_size` - it looks much more readable.